### PR TITLE
Added Respective Models, Controllers and Associations

### PR DESCRIPTION
**Pull request by UsamaShahzad24789**
I have generated models, controllers required for our project and created respective associations between them.
Following are the models existing:

- Account[Devise Model] (has enum for role {admin,user}  Association: has one profile
- Profile   {has enum for status{public,private} Association: has many posts
- Post.     Association: has many comments, has many likes
- Comment
- Like

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,6 +6,14 @@ ruby "3.2.2"
 # Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
 gem "rails", "~> 7.0.6"
 
+gem "devise"
+
```
Comment (ayzahamid) [inline Gemfile:10]: Please remove extra lines between gems.
Install rubocop and fix all the offenses.

File: app/controllers/accounts/confirmations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  # The path used after resending confirmation instructions.
+  # def after_resending_confirmation_instructions_path_for(resource_name)
+  #   super(resource_name)
+  # end
+
+  # The path used after confirmation.
+  # def after_confirmation_path_for(resource_name, resource)
+  #   super(resource_name, resource)
+  # end
```
Comment (ayzahamid) [inline app/controllers/accounts/confirmations_controller.rb:29]: What is this file doing?

File: app/controllers/accounts/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  #   super
+  # end
+
+  # protected
+
+  # The path used when OmniAuth fails
+  # def after_omniauth_failure_path_for(scope)
+  #   super(scope)
+  # end
+end
```
Comment (ayzahamid) [inline app/controllers/accounts/omniauth_callbacks_controller.rb:30]: What is this file doing?

File: app/controllers/accounts/passwords_controller.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Accounts::PasswordsController < Devise::PasswordsController
+  # GET /resource/password/new
```
Comment (ayzahamid) [inline app/controllers/accounts/passwords_controller.rb:4]: What is this file doing?

File: app/controllers/accounts/registrations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  # before_action :configure_account_update_params, only: [:update]
+
+  GET /resource/sign_up
+  def new
+    super
+  end
+
+  # POST /resource
+  def create
+    debugger
```
Comment (ayzahamid) [inline app/controllers/accounts/registrations_controller.rb:14]: 🚫🚫 🚫 🚫 🚫

File: app/controllers/accounts/registrations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  # removing all OAuth session data.
+  # def cancel
+  #   super
+  # end
+
+  # protected
+
+  # If you have extra params to permit, append them to the sanitizer.
+  # def configure_sign_up_params
+  #   devise_parameter_sanitizer.permit(:sign_up, keys: [:attribute])
```
Comment (ayzahamid) [inline app/controllers/accounts/registrations_controller.rb:46]: Why do we have so much commented code?

File: app/controllers/comments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+class CommentsController < ApplicationController
```
Comment (ayzahamid) [inline app/controllers/comments_controller.rb:1]: What is this doing?

File: app/controllers/posts_controller.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+class PostsController < ApplicationController
```
Comment (ayzahamid) [inline app/controllers/posts_controller.rb:1]: What is this doing?

File: app/controllers/profile_controller.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+class ProfileController < ApplicationController
```
Comment (ayzahamid) [inline app/controllers/profile_controller.rb:1]: What is this doing?

File: app/helpers/comments_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (ayzahamid) [inline app/helpers/comments_helper.rb:1]: What is this doing?

File: app/helpers/comment_helper.rb [hunk 11]
```diff
@@ -0,0 +1,2 @@
+module CommentHelper
```
Comment (ayzahamid) [inline app/helpers/comment_helper.rb:1]: What is this doing?

File: app/helpers/home_helper.rb [hunk 12]
```diff
@@ -0,0 +1,2 @@
+module HomeHelper
```
Comment (ayzahamid) [inline app/helpers/home_helper.rb:1]: What is this doing?

File: app/helpers/posts_helper.rb [hunk 13]
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
```
Comment (ayzahamid) [inline app/helpers/posts_helper.rb:1]: What is this doing?

File: app/helpers/profile_helper.rb [hunk 14]
```diff
@@ -0,0 +1,2 @@
+module ProfileHelper
```
Comment (ayzahamid) [inline app/helpers/profile_helper.rb:1]: What is this doing?

File: app/models/comment.rb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+class Comment < ApplicationRecord
```
Comment (ayzahamid) [inline app/models/comment.rb:1]: What is this doing?

File: app/models/like.rb [hunk 16]
```diff
@@ -0,0 +1,2 @@
+class Like < ApplicationRecord
```
Comment (ayzahamid) [inline app/models/like.rb:1]: What is this doing?

File: app/models/post.rb [hunk 17]
```diff
@@ -0,0 +1,4 @@
+class Post < ApplicationRecord
+  has_many :comments,dependent: :destroy
+  has_many :likes,dependent: :destroy
```
Comment (ayzahamid) [inline app/models/post.rb:3]: Space missing after comma
```suggestion
  has_many :comments, dependent: :destroy
  has_many :likes, dependent: :destroy
```

File: app/views/devise/confirmations/new.html.erb [hunk 18]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
+
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
```
Comment (ayzahamid) [inline app/views/devise/confirmations/new.html.erb:7]: Avoid using br tags. Implement styling using proper bootstrap classes

File: app/views/devise/registrations/new.html.erb [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+    <%# <% if @minimum_password_length %>
+    <%# <em>(<%= @minimum_password_length characters minimum)</em> %>
+    <%# <% end<br /> %>
+    <%# <%= f.password_field :password, autocomplete: "new-password" %>
+  <%# </div> %>
+
+  <%# <div class="field"> %>
+    <%# <%= f.label :password_confirmation<br /> %>
+    <%# <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  <%# </div> %>
```
Comment (ayzahamid) [inline app/views/devise/registrations/new.html.erb:22]: Again we have commented code.

File: test/controllers/comment_controller_test.rb [hunk 20]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class CommentControllerTest < ActionDispatch::IntegrationTest
```
Comment (ayzahamid) [inline test/controllers/comment_controller_test.rb:3]: What is this file doing?

File: test/controllers/comments_controller_test.rb [hunk 21]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class CommentsControllerTest < ActionDispatch::IntegrationTest
```
Comment (ayzahamid) [inline test/controllers/comments_controller_test.rb:3]: What is this file doing?

File: test/controllers/home_controller_test.rb [hunk 22]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class HomeControllerTest < ActionDispatch::IntegrationTest
```
Comment (ayzahamid) [inline test/controllers/home_controller_test.rb:3]: What is this file doing?

File: test/controllers/posts_controller_test.rb [hunk 23]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class PostsControllerTest < ActionDispatch::IntegrationTest
```
Comment (ayzahamid) [inline test/controllers/posts_controller_test.rb:3]: What is this file doing?

File: test/controllers/profile_controller_test.rb [hunk 24]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class ProfileControllerTest < ActionDispatch::IntegrationTest
```
Comment (ayzahamid) [inline test/controllers/profile_controller_test.rb:3]: What is this file doing?

File: test/fixtures/accounts.yml [hunk 25]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the "{}" from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
```
Comment (ayzahamid) [inline test/fixtures/accounts.yml:7]: What is this file doing?

File: test/fixtures/comments.yml [hunk 26]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the "{}" from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
```
Comment (ayzahamid) [inline test/fixtures/comments.yml:7]: What is this file doing?

File: test/fixtures/likes.yml [hunk 27]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the "{}" from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
```
Comment (ayzahamid) [inline test/fixtures/likes.yml:7]: What is this file doing?

File: test/fixtures/posts.yml [hunk 28]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the "{}" from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
```
Comment (ayzahamid) [inline test/fixtures/posts.yml:7]: What is this file doing?

File: test/fixtures/profiles.yml [hunk 29]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the "{}" from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
```
Comment (ayzahamid) [inline test/fixtures/profiles.yml:7]: What is this file doing?

File: test/models/comment_test.rb [hunk 30]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class CommentTest < ActiveSupport::TestCase
```
Comment (ayzahamid) [inline test/models/comment_test.rb:3]: What is this file doing?

File: test/models/account_test.rb [hunk 31]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class AccountTest < ActiveSupport::TestCase
```
Comment (ayzahamid) [inline test/models/account_test.rb:3]: What is this file doing?

File: test/models/like_test.rb [hunk 32]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class LikeTest < ActiveSupport::TestCase
```
Comment (ayzahamid) [inline test/models/like_test.rb:3]: What is this file doing?

File: test/models/post_test.rb [hunk 33]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class PostTest < ActiveSupport::TestCase
```
Comment (ayzahamid) [inline test/models/post_test.rb:3]: What is this file doing?

File: app/controllers/accounts/registrations_controller.rb [hunk 34]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class Accounts::RegistrationsController < Devise::RegistrationsController
+  # before_action :configure_sign_up_params, only: [:create]
+  # before_action :configure_account_update_params, only: [:update]
+
+  GET /resource/sign_up
```
Comment (ayzahamid) [inline app/controllers/accounts/registrations_controller.rb:7]: `GET /resource/sign_up`
What is this doing?

**Review by ayzahamid — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (UsamaShahzad24789) [reply Gemfile:10]: Sure, I will fix all the offense using robocop

File: app/controllers/accounts/confirmations_controller.rb (hunk 2 above)
Comment (UsamaShahzad24789) [reply app/controllers/accounts/confirmations_controller.rb:29]: All the files under controller/accounts/ are generated by devise , these files are provided in order to add any extra functionality to devise SignIn/SignUp Flow. These files stays commented unless we want to add any user desired functionality in these controllers as they are inherited from the Devise Controllers.
These controllers were generated using 
rails generate devise:controllers accounts

File: app/controllers/accounts/omniauth_callbacks_controller.rb (hunk 3 above)
Comment (UsamaShahzad24789) [reply app/controllers/accounts/omniauth_callbacks_controller.rb:30]: **These controllers will be used during the signUp flow , you will come along the code in the up coming PR**

All the files under controller/accounts/ are generated by devise , these files are provided in order to add any extra functionality to devise SignIn/SignUp Flow. These files stays commented unless we want to add any user desired functionality in these controllers as they are inherited from the Devise Controllers.
These controllers were generated using
rails generate devise:controllers accounts

File: test/models/comment_test.rb (hunk 30 above)
Comment (UsamaShahzad24789) [reply test/models/comment_test.rb:3]: These test files were automatically generated when I generate a controller from terminal . These files will be removed.

File: test/models/account_test.rb (hunk 31 above)
Comment (UsamaShahzad24789) [reply test/models/account_test.rb:3]: These test files were automatically generated when I generate a controller from terminal . These files will be removed.

File: test/models/like_test.rb (hunk 32 above)
Comment (UsamaShahzad24789) [reply test/models/like_test.rb:3]: These test files were automatically generated when I generate a controller from terminal . These files will be removed.

File: test/models/post_test.rb (hunk 33 above)
Comment (UsamaShahzad24789) [reply test/models/post_test.rb:3]: These test files were automatically generated when I generate a controller from terminal . These files will be removed.

File: app/controllers/accounts/passwords_controller.rb (hunk 4 above)
Comment (UsamaShahzad24789) [reply app/controllers/accounts/passwords_controller.rb:4]: All the files under controller/accounts/ are generated by devise , these files are provided in order to add any extra functionality to devise SignIn/SignUp Flow. These files stays commented unless we want to add any user desired functionality in these controllers as they are inherited from the Devise Controllers.
These controllers were generated using
rails generate devise:controllers accounts

File: app/controllers/accounts/registrations_controller.rb (hunk 6 above)
Comment (UsamaShahzad24789) [reply app/controllers/accounts/registrations_controller.rb:46]: All the files under controller/accounts/ are generated by devise , these files are provided in order to add any extra functionality to devise SignIn/SignUp Flow. These files stays commented unless we want to add any user desired functionality in these controllers as they are inherited from the Devise Controllers.
These controllers were generated using
rails generate devise:controllers accounts

**Pull request closed without merging**